src/java/com/google/cloud/bigquery/dwhassessment/hooks/logger/EventRecordConstructor.java

```diff
@@ -191,13 +194,23 @@ private static String dumpPerfData(PerfLogger perfLogger) {
   }
 
   private static List<String> getTablesFromEntitySet(Set<? extends Entity> entities) {
-    List<String> tableNames = new ArrayList<>();
+    Set<String> tableNames = new HashSet<>();
     for (Entity entity : entities) {
-      if (entity.getType() == TABLE || entity.getType() == PARTITION) {
+      if (entity.getType() == TABLE) {
         tableNames.add(entity.getTable().getCompleteName());
       }
     }
-    return tableNames;
+    return new ArrayList<>(tableNames);
+  }
+
+  private static List<String> getPartitionsFromEntitySet(Set<? extends Entity> entities) {
+    Set<String> partitionNames = new HashSet<>();
+    for (Entity entity : entities) {
+      if (entity.getType() == PARTITION) {
+        partitionNames.add(entity.getPartition().getName());
```

Comment:
Is partition name fully qualified (i.e. contains database name and table name) or not? If the second, it will be harder to distinguish partitions from different tables, e.g. in query with joins. If so, consider constructing a fully qualified string, which can be parsed on the assessment side.